src/Model/ModelManagerInterface.php

```diff
@@ -25,70 +26,52 @@
 interface ModelManagerInterface extends DatagridManagerInterface
 {
     /**
-     * @param string $class
-     * @param string $name
-     *
-     * @return FieldDescriptionInterface
+     * @param array<string, mixed> $options
      */
-    public function getNewFieldDescriptionInstance($class, $name, array $options = []);
+    public function getNewFieldDescriptionInstance(string $class, string $name, array $options = []): FieldDescriptionInterface;
 
     /**
-     * @param object $object
-     *
      * @throws ModelManagerException
      */
-    public function create($object);
+    public function create(object $object): void;
 
     /**
-     * @param object $object
-     *
      * @throws ModelManagerException
      */
-    public function update($object);
+    public function update(object $object): void;
 
     /**
-     * @param object $object
-     *
      * @throws ModelManagerException
      */
-    public function delete($object);
+    public function delete(object $object): void;
 
     /**
-     * @param string $class
+     * @param array<string, mixed> $criteria
      *
-     * @return array all objects matching the criteria
+     * @return object[] all objects matching the criteria
      */
-    public function findBy($class, array $criteria = []);
+    public function findBy(string $class, array $criteria = []): array;
```

Comment:
by the way, you can do something like this

```php
/**
 * @template T
 * @param class-string<T> $class
 * @return list<T>
 */
```

Reply:
I can search for cases where templates can be used if you want. But I think you can find them by your self

Reply:
Fixed